src/lib/timing.c

```diff
@@ -125,6 +126,51 @@ static uint64_t xdebug_get_nanotime_rel(xdebug_nanotime_context *nanotime_contex
 }
 #endif
 
+#if WIN_SUPPORTS_RDTSC
+static inline uint64_t xdebug_get_rdtsc()
+{
+	return __rdtsc();
+}
+
+static void xdebug_nanotime_init_rdtsc_to_nanos()
+{
+	/* Calibrate TSC scale, wait at least 2 ms */
+	do {
+		__nop();
+	} while (xdebug_get_nanotime() - context.start_abs < 2 * NANOS_IN_MILLISEC);
+	XG_BASE(nanotime_context).rdtsc_to_nanos = (xdebug_get_nanotime() - context.start_abs)
+		/ (double)(xdebug_get_rdtsc() - context.start_rdtsc);
+}
+#endif
+
+static void xdebug_nanotime_init_get_nanotime_func_overhead()
+{
+	int i, j;
+	uint64_t t, t_last, t_diff_sum;
+	unsigned int t_diff_min;
+
+	XG_BASE(nanotime_context).get_nanotime_func_overhead = NANOTIME_MIN_STEP - NANOTIME_MIN_STEP_EXCL_OVERHEAD;
+
+	/* 10k iterations total, average overhead is 30 ns per get nanotime call,
+	 * thus total time is about 0.3 ms */
+	t_diff_min = UINT_MAX;
+	for (i = 0; i < 100; i++) {
+		t_diff_sum = 0;
+		t_last = xdebug_get_nanotime();
+		for (j = 0; j < 100; j++) {
+			t = xdebug_get_nanotime();
+			t_diff_sum += (t - t_last);
+			t_last = t;
+		}
+
+		if (t_diff_sum / j < t_diff_min) {
+			t_diff_min = t_diff_sum / j;
+		}
+	}
+
+	XG_BASE(nanotime_context).get_nanotime_func_overhead = t_diff_min;
```

Comment:
I'm not inclined to merge anything that requires a loop for calibration.

Reply:
This PR is based on top of https://github.com/xdebug/xdebug/pull/629 which must be finished first as mentioned in the description. Calliabration is needed, but we can make it optional, see https://github.com/xdebug/xdebug/pull/629/files#r566760005.